revolv/payments/utils.py

```diff
@@ -1,6 +1,398 @@
+import datetime
+import urllib
+
+from django.db.models import Sum
+from revolv.payments.models import Payment, AdminReinvestment, AdminRepayment, AdminAdjustment
+from revolv.project.models import Project
+
 class NotEnoughFundingException(Exception):
     pass
 
 
 class ProjectNotCompleteException(Exception):
     pass
+
+class AccountingAggregator():
+    
+    """
+    An aggregator class that does much of the computation for the AdministratorAccountingView
+    view in revolv.administrator.views.
+    This class contains a bunch of helper functions and one main function (aggregate) which returns 
+    a dictionary of relevant accounting data, filtered by instance variables representing the start date,
+    end date, and project filters.
+    """
+
+    def __init__(self, start_date=None, end_date=None, project_filter=None, revolv_earnings_rate=0, payment_service_fee_rate=0):
+        # to prevent uneccessary operations
```

Comment:
If you use project ids, then you won't have to clean the project names, which means you'll always want to instantiate this with all the arguments, which means you won't need this if clause. Also, couldn't you technically want to instantiate an aggregator with no start date? That case would be a big bug, because nothing else would get instantiated even if you passed it in.